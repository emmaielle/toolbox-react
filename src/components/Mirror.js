import React, { Component } from 'react'
import autoBind from 'auto-bind';
import Loader from 'react-loaders'
import { Col } from 'react-bootstrap';
import _ from 'lodash';

export class Mirror extends Component {
  
    constructor(props){
        super(props);
        this.state = {
            text: ''
        };
        autoBind(this);
        this.change = this.props.changeInput;
    }

    handleChangeInput(ev){
        const text = ev.target.value;
        this.setState({text}, () => {
            if (text){
                this.change(text);
            }
        })
        
        
    }

    render() {
        debugger;
        const { mirrored } = this.props;
        let renderBody = (
            <div className="success-box">
            </div>
            );
        if (mirrored){
            const { loading, error, mirror } = mirrored;
            if (error){
                renderBody = <p>There has been an unexpected error</p>;
            }
            else if (mirror){
                renderBody = (
                    <Col xs={12} sm={10} smOffset={1} md={8} mdOffset={2} lg={8} lgOffset={2} className="success-box">
                        <div className="text title">Result:</div>                        
                        <div className="text">{mirror.text}</div>
                    </Col>
                );
            }
        }

        return (
        <div className="mirror">
            <div>
                <textarea className="text-area"
                placeholder="Enter a text" 
                value={this.state.text} 
                onChange={this.handleChangeInput} />
            </div>
            <div className="result-container">
                {renderBody}
            </div>
        </div>
        )
    }
}

export default Mirror
