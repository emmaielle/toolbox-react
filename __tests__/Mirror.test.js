// MyComponent.test.js
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Mirror from '../src/components/Mirror';

describe("Mirror component", () => {
  it("should render the Mirror component", () => {
    const wrapper = shallow(<Mirror />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});