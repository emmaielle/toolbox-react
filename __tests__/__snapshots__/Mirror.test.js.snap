// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mirror component should render the Mirror component 1`] = `
<div
  className="mirror"
>
  <div>
    <textarea
      className="text-area"
      onChange={[Function]}
      placeholder="Enter a text"
      value=""
    />
  </div>
  <div
    className="result-container"
  >
    <div
      className="success-box"
    />
  </div>
</div>
`;
